CMakeLists.txt

```diff
@@ -7,10 +7,13 @@ endif()
 
 # Determine if fmt is built as a subproject (using add_subdirectory)
 # or if it is the master project.
-set(MASTER_PROJECT OFF)
-if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
-  set(MASTER_PROJECT ON)
-  message(STATUS "CMake version: ${CMAKE_VERSION}")
+if(NOT DEFINED FMT_MASTER_PROJECT)
+  if (CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+    set(FMT_MASTER_PROJECT ON)
+    message(STATUS "CMake version: ${CMAKE_VERSION}")
+  else ()
+    set(FMT_MASTER_PROJECT OFF)
```

Comment:
I suggest setting `FMT_MASTER_PROJECT` to `OFF` outside of if as before and removing the else branch.